### PR TITLE
feat: upgraded to the latest version of appsflyer sdk and added support for new events

### PR DESCRIPTION
**Pull request by desusai7**
# Description

* Upgraded Rudder SDK Version used in the Device Mode to be '~> 1.12'
* Upgraded the AppsFlyer SDK version used in the Device Mode to be  '~> 6.10'
* Added the Capability for the User to override the versions of both Rudder SDK and Appsflyer SDK in case of Conflicts.
* Added Mapping for a few new events like Ad View, Ad Click, Add Payment Info, Add to Wishlist, Share, Rate
* Sending Custom properties for all the events and the standard ones supported by AppsFlyer.
* Updated Sample App to read write key and data plane from .plist
* Updated the Sample App to have all the E-commerce Events instrumented for easier testing
* Updated the Sample App to wait for the user consent and then initialize the AppsFlyer SDK (includes requesting consent from end-user)

### CODE REVIEW

File: Rudder-Appsflyer.podspec [hunk 1]
```diff
@@ -12,7 +15,7 @@ Rudder is a platform for collecting, storing and routing customer event data to
                        DESC
 
   s.homepage         = 'https://github.com/rudderlabs/rudder-integration-appsflyer-ios'
-  s.license          = { :type => "Apache", :file => "LICENSE" }
+  s.license          = { :type => "Apache", :file => "LICENSE.md" }
   s.author           = { 'RudderStack' => 'arnab@rudderlabs.com' }
```
Comment (pallabmaiti) [inline Rudder-Appsflyer.podspec]:
```suggestion
  s.author           = { 'RudderStack' => 'arnab@rudderstack.com' }
```

File: Rudder-Appsflyer.podspec
Comment (pallabmaiti) [inline Rudder-Appsflyer.podspec:1]: Remove `s.pod_target_xcconfig` and `s.user_target_xcconfig`

File: Rudder-Appsflyer.podspec [hunk 2]
```diff
@@ -12,7 +15,7 @@ Rudder is a platform for collecting, storing and routing customer event data to
                        DESC
 
   s.homepage         = 'https://github.com/rudderlabs/rudder-integration-appsflyer-ios'
-  s.license          = { :type => "Apache", :file => "LICENSE" }
+  s.license          = { :type => "Apache", :file => "LICENSE.md" }
```
Comment (pallabmaiti) [inline Rudder-Appsflyer.podspec:18]: The `LICENSE` file doesn't have `md` extension.

File: Rudder-Appsflyer/Classes/RudderAppsflyerIntegration.m [hunk 3]
```diff
@@ -156,37 +209,37 @@ - (void) addCheckoutProperties: (NSDictionary *) properties params: (NSMutableDi
...
-        NSString *currency = properties[@"currency"];
+        NSString *currency = properties[KeyCurrency];
         if (currency != nil) {
             [params setValue:currency forKey:AFEventParamCurrency];
         }
         
-        NSString *orderId = properties[@"order_id"];
+        NSString *orderId = properties[KeyOrderId];
         if (orderId != nil) {
             [params setValue:orderId forKey:AFEventParamReceiptId];
```
Comment (pallabmaiti) [inline Rudder-Appsflyer/Classes/RudderAppsflyerIntegration.m:219]: Shouldn't `AFEventParamReceiptId` be `transaction_id`?

**Review by pallabmaiti — COMMENTED** [review]
Remove these

1. [Assets](https://github.com/rudderlabs/rudder-integration-appsflyer-ios/tree/master/Rudder-Appsflyer/Assets) folder
2. [.gitkeep](https://github.com/rudderlabs/rudder-integration-appsflyer-ios/blob/master/Rudder-Appsflyer/Classes/.gitkeep)

File: Rudder-Appsflyer/Classes/RudderAppsflyerIntegration.m (hunk 3 above)
Comment (desusai7) [reply Rudder-Appsflyer/Classes/RudderAppsflyerIntegration.m:219]: No, AppsFlyer standard params say that this should be linked to af_order_id and af_receipt_id, please refer [here](https://support.appsflyer.com/hc/en-us/articles/360019347957#purchase-af_purchase) for more details about the same

Comment (pallabmaiti) [reply Rudder-Appsflyer/Classes/RudderAppsflyerIntegration.m:219]: Cool.

**Comment by desusai7** [comment]
@pallabmaiti, I addressed all the comments, can you please retake a look?

File: Rudder-Appsflyer.podspec [hunk 4]
```diff
@@ -12,23 +15,32 @@ Rudder is a platform for collecting, storing and routing customer event data to
                        DESC
 
   s.homepage         = 'https://github.com/rudderlabs/rudder-integration-appsflyer-ios'
-  s.license          = { :type => "Apache", :file => "LICENSE" }
-  s.author           = { 'RudderStack' => 'arnab@rudderlabs.com' }
+  s.license          = { :type => "Apache", :file => "LICENSE.md" }
+  s.author           = { 'RudderStack' => 'sdk@rudderstack.com' }
```
Comment (pallabmaiti) [inline Rudder-Appsflyer.podspec:19]: Do we have this email? I believe we have `sdk-accounts@rudderstack.com`.

Comment (desusai7) [reply Rudder-Appsflyer.podspec:19]: yes, we do have this

**Review by pallabmaiti — APPROVED** [review]
Please squash the commits.

**Comment by desusai7** [comment]
Closed this as part of branch renaming for the CI

**Pull request closed without merging**